common/src/main/java/tachyon/security/UserInformation.java

```diff
@@ -40,11 +48,35 @@
   private static final boolean IS_64_BIT = PlatformUtils.PROCESSOR_BIT.contains("64");
   private static final boolean AIX = PlatformUtils.OS_NAME.equals("AIX");
 
+  /** The configuration to use */
+  private static TachyonConf sConf;
+  /** The authentication method to use */
+  private static AuthenticationFactory.AuthTypes sAuthType;
+  /** UserInformation about the login user */
+  private static UserInformation sLoginUser;
+
+  /**
+   * The subject of this UserInformation instance. A subject contains several perspective of a
+   * user.
+   */
+  private final Subject mSubject;
+  /** The Tachyon user represented by this UserInformation instance. */
+  private final User mUser;
+
   static {
     OS_LOGIN_MODULE_NAME = getOSLoginModuleName();
     OS_PRINCIPAL_CLASS_NAME = findOsPrincipalClassName();
   }
 
+  /**
+   * Create a UserInformation for the given subject.
+   * @param subject the user's subject
+   */
+  UserInformation(Subject subject) {
+    mSubject = subject;
+    mUser = subject.getPrincipals(User.class).iterator().next();
```

Comment:
Thanks. Add check on it. Since the structure is changed, the check is added in `UserInformation.login()` method.